wrapanapi/systems/nuage.py

```diff
@@ -24,75 +19,61 @@ class NuageSystem(System):
+        else:
+            from vspk import v5_0 as vspk
+        return vspk
 
     @property
-    def auth(self):
-        if not self._auth:
-            login_url = self.url + "/me"
-            response = requests.request('get', login_url, auth=HTTPBasicAuth(*self.login_auth),
-                headers=self.common_headers, verify=False)
-            response.raise_for_status()
-            r = response.json()[0]
-            self._auth = (r.get('userName'), r.get('APIKey'))
-        return self._auth
+    def api(self):
+        if self._api is None:
+            session = self.vspk.NUVSDSession(
+                username=self.username,
+                password=self.password,
+                enterprise=self.enterprise,
+                api_url=self.url
+            )
+            session.start()
+            self._api = session.user
+        return self._api
+
+    def disconnect(self):
+        self._api = None
 
     @property
-    def common_headers(self):
-        return {
-            'Content-Type': 'application/json; charset=UTF-8',
-            'X-Nuage-Organization': 'csp'
-        }
-
-    def list_cloud_subnets(self):
-        return self._request_list('/subnets', 'get', exclude_name='BackHaulSubnet')
-
-    def list_security_groups(self):
-        return self._request_list('/policygroups', 'get')
-
-    def list_enterprises(self):
-        return self._request_list('/enterprises', 'get')
-
-    def list_domains(self):
-        return self._request_list('/domains', 'get')
+    def _identifying_attrs(self):
+        return {'url': self.url}
 
-    def _request_list(self, *args, **kwargs):
-        resp = self._request(*args, **kwargs)
-        return [] if resp is None else resp
+    def info(self):
+        return 'NuageSystem: url={}'.format(self.url)
 
-    def _request(self, url, method, data=None, exclude_name=None):
-        headers = self.common_headers
-        if exclude_name:
-            headers.update({
-                'X-Nuage-FilterType': 'predicate',
-                'X-Nuage-Filter': "name ISNOT '{}'".format(exclude_name),
-            })
+    def list_floating_network_resources(self):
+        return self.api.shared_network_resources.get(filter='type is "FLOATING"')
 
-        response = requests.request(
-            method, self.url + url,
-            auth=HTTPBasicAuth(*self.auth),
-            headers=headers,
-            verify=False,
-            data=json.dumps(data) if data else None  # workaround Nuage bug that empty body "" is
-        )                                            # returned when emtpy JSON "{}" should be
-        response.raise_for_status()
-        return response.json() if response.text else None
+    def list_vports(self):
+        vports = [d.vports.get() and d.vports for d in self.api.domains.get()]
+        vports.extend([d.vports.get() and d.vports for d in self.api.l2_domains.get()])
+        return [vport for sublist in vports for vport in sublist]
```

Comment:
Please include in-code comment about why the list comprehensions are using `and`, and then the return is re-iterating the list. Looks like first two are generating lists of booleans?

Reply:
It's strange because you need to explicitly fetch objects before accessing them. But hey, who am I to judge? :) From what I've tried, the first two calls always return some kind of empty iterator (whuch is resolved as truthful), but they do set the instance attribute. So we can use `and` to then access it in the same single line. Can't deny it's a smart trick, can you? 😎